File: C# Assignment/assignment solutions/ass4/1/2/Emp.cs
using System;


namespace _2
{
    abstract class Emp:Person
    {
        int id;
        protected double basic;
        public Emp()
        {
            id = 0;
            basic = 0;
        }

        public Emp(string nm,int d,int m,int y,int i, int b):base(nm,d,m,y)
        {
            id = i;
            basic = b;
        }
        public abstract double calsal();
        
        public virtual new void accept()
        {
            Console.WriteLine("Enter the id");
            id=int.Parse(Console.ReadLine());
            Console.WriteLine("Enter the salary");
            basic=int.Parse(Console.ReadLine());
            base.accept();
        }

        public virtual new void display()
        {
            
            Console.WriteLine("Id:"+id);
            Console.WriteLine("Basic salary:"+basic);
            base.display();
        }

    }
}

File: Class Demo/VB.Net/OOPsConcept/OOPsConcept/Emp.vb
'MustInherit for abstract class
Public MustInherit Class Emp
    Dim name As String
    Dim eid As Integer
    Protected basic As Double
    'new keyword is used to create constructor
    Public Sub New()
        name = ""
        eid = 0
        basic = 0
    End Sub
    Public Sub New(ByVal e As Integer, ByVal nm As String,
                   ByVal bs As Double)
        name = nm
        eid = e
        basic = bs
    End Sub
    'Overridable is used in place of virtual
    Public Overridable Sub Display()
        Console.WriteLine("Name=" + name)
        Console.WriteLine("Eid=" + eid.ToString())
        Console.WriteLine("Basic=" + basic.ToString())
    End Sub


    'Public Overridable Function CalSal() As Double
    '    Return basic
    'End Function

    'Mustoverride in place of Abstract
    Public MustOverride Function CalSal() As Double

End Class
